apps/server/src/modules/tool/common/service/validation/rules/parameter-entry-value-validator.ts

```diff
@@ -10,7 +15,10 @@ export class ParameterEntryValueValidator implements ParameterEntryValidator {
 		toolId: EntityId | undefined
 	): ValidationError[] {
 		if (entry.value === undefined || entry.value === '') {
-			return [new ToolParameterValueMissingLoggableException(toolId, declaration)];
+			if (declaration.isOptional) {
+				return [new ToolParameterOptionalValueMissingLoggableException(toolId, declaration)];
```

Comment:
So, the user inst allowed to start a tool with an optional parameter? Then the question: Why we have mandatory and optional parameters?